Sources/SwiftGraphQLClient/Client/Core.swift

```diff
@@ -255,7 +255,21 @@ public class Client: GraphQLClient, ObservableObject {
         
         return self.execute(operation: operation)
     }
-    
+
+    /// Executes a query request with given execution parameters.
+    ///
+    /// Note: While this behaves much the same as the published-based
+    /// APIs, async/await inherently does __not__ support multiple
+    /// return values. If you expect multiple values from an async/await
+    /// API, please use the corresponding publisher API instead.
```

Comment:
Let's also add a comment that cache invalidation does not work in this case.

The main benefit of publishers is that when another query or mutation invalidates the query we can simply use the new value which is not the case with asynchronous queries that only serve one response.

I think this is also only applicable to queries. Mutations work far more predictably with async/await methods.

Reply:
Yep, I agree. I'll do that. To be honest that's something that we can improve more considerably when there's a deeper concurrency implementation. But you're right well we're just wrapping combined with changing/losing some behaviour with his API. But it's a nice balance for the convenience I think, and we're not currently deprecating the old API anyway👍